File: Searching and Sorting Algorithms/16QuickSort.py
def quick_sort(arr, low, high):
    # Write your code here

    if low>=high:
        return
    
    pivot=partition(arr,low,high)
    quick_sort(arr,low,pivot-1)
    quick_sort(arr,pivot+1,high)



def partition(arr, low, high):
    pivot=arr[high]
    i=low-1
    for j in range(low,high):
        if arr[j]<pivot:
            temp=arr[j]
            i+=1
            arr[j]=arr[i]
            arr[i]=temp
    i+=1
    temp=arr[i]
    arr[i]=pivot
    arr[high]=temp
    return i

arr=[9,8,7,6,5,4,3,2,1]
quick_sort(arr,0,len(arr)-1)
print(arr)

File: Searching and Sorting Algorithms/codigo_cs/16QuickSort.cs
using System;


public class Test16

{

    public static void QuickSort(int[] arr, int low, int high)
    {
        // write your code here
        if (low >= high)
        {
            return;
        }
        int pivot = Partition(arr, low, high);
        QuickSort(arr, low, pivot - 1);
        QuickSort(arr, pivot + 1, high);
    }

    static int Partition(int[] arr, int low, int high)
    {
        // write your code here
        int pivot = arr[high];
        int i = low - 1;
        for (int j = low; j < high; j++)
        {
            if (arr[j] < pivot)
            {
                i++;
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
        i++;
        int temp2 = arr[i];
        arr[i] = pivot;
        arr[high] = temp2;
        return i;

    }

}